### PR TITLE
Fix conversion in ssl* files and in internal.c

### PR DESCRIPTION
**Pull request by gasbytes**
Fix issues related to -Wconversion and -Wsign-conversion in the ssl* files and in internal.c.

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -7894,7 +7894,7 @@ int AllocKey(WOLFSSL* ssl, int type, void** pKey)
     }
 
     /* Allocate memory for key */
-    *pKey = (void *)XMALLOC(sz, ssl->heap, type);
+    *pKey = (void *)XMALLOC((size_t)sz, ssl->heap, type);
```
Comment (bandi13) [inline src/internal.c:7897]: Better to change the type of the variable declaration.

File: src/internal.c [hunk 2]
```diff
@@ -10697,8 +10697,8 @@ int SendBuffered(WOLFSSL* ssl)
             return SEND_OOB_READ_E;
         }
 
-        ssl->buffers.outputBuffer.idx += sent;
-        ssl->buffers.outputBuffer.length -= sent;
+        ssl->buffers.outputBuffer.idx += (word32)sent;
```
Comment (bandi13) [inline src/internal.c:10700]: Change variable type instead

File: src/internal.c [hunk 3]
```diff
@@ -10821,7 +10821,7 @@ int GrowInputBuffer(WOLFSSL* ssl, int size, int usedLength)
         return BAD_FUNC_ARG;
     }
 
-    tmp = (byte*)XMALLOC(size + usedLength + align,
+    tmp = (byte*)XMALLOC((size_t)(size + usedLength + align),
```
Comment (bandi13) [inline src/internal.c:10824]: I wonder if it is possible to change the function input variable types

File: src/internal.c [hunk 4]
```diff
@@ -20727,7 +20727,7 @@ static int DoAlert(WOLFSSL* ssl, byte* input, word32* inOutIdx, int* type)
             if (CipherHasExpIV(ssl))
                 ivExtra = AESGCM_EXP_IV_SZ;
         }
-        dataSz -= ivExtra;
+        dataSz -= (word32)ivExtra;
```
Comment (bandi13) [inline src/internal.c:20730]: Change variable type instead

File: src/internal.c [hunk 5]
```diff
@@ -20454,7 +20454,7 @@ int DoApplicationData(WOLFSSL* ssl, byte* input, word32* inOutIdx, int sniff)
             if (dataSz < 0) return dataSz;
         }
 #endif
-        idx += rawSz;
+        idx += (word32)rawSz;
```
Comment (bandi13) [inline src/internal.c:20457]: Change variable type instead

**Review by bandi13 — CHANGES_REQUESTED** [review]
Please go through and check if the variable declaration type should be changed instead of being cast.

**Review by bandi13 — APPROVED** [review]

**Review by douzzer — APPROVED** [review]

**Pull request merged**